frontend/lib/wallets/Coinbase.ts

```diff
@@ -0,0 +1,81 @@
+import CoinbaseWalletSDK, { CoinbaseWalletProvider } from '@coinbase/wallet-sdk';
+import { ethers } from 'ethers';
+import { getChainIdByNetworkType, getNetworkTypeByChainId } from 'auctions-core/src/network';
+import { setSigner } from 'auctions-core/src/signer';
+import { formatToHexWithoutPad } from 'auctions-core/helpers/format';
+import CoinbaseLogo from '~/assets/icons/wallets/coinbase.svg';
+import AbstractWallet from '~/lib/wallets/AbstractWallet';
+
+export default class Coinbase extends AbstractWallet {
+    public static title = 'Coinbase Wallet';
+    public static icon = CoinbaseLogo;
+    public static downloadUrl = 'https://www.coinbase.com/wallet/downloads';
+
+    private static provider?: CoinbaseWalletProvider;
+
+    private addresses?: string[];
+
+    public get address() {
+        if (Array.isArray(this.addresses) && this.addresses[0]) {
+            return this.addresses[0].toLowerCase();
+        }
+    }
+
+    getProvider(): ethers.providers.JsonRpcProvider {
+        const rpcUrl = process.env.RPC_URL;
+        if (!rpcUrl) {
+            throw new Error(`No RPC_URL env variable was provided`);
+        }
+        if (!Coinbase.provider) {
+            const walletSdk = new CoinbaseWalletSDK({ appName: 'Unified Auctions UI' });
```

Comment:
Maybe you can provide [SITE_TITLE from the nuxt config](https://github.com/sidestream-tech/unified-auctions-ui/blob/6cdba5def95207b3e1f4178878c1678124b18d63/frontend/nuxt.config.js#L4) via env vars if you don't want to hardcode it here